ImageBuilderRAM/ImagePipeline_SharedAccount_Sample_AmznLinux.yml

```diff
@@ -0,0 +1,387 @@
+                # Nothing to delete:
+                response_status = cfnresponse.SUCCESS
+              elif event['RequestType'] in ('Create', 'Update'):
+                resource_properties = event['ResourceProperties']
+                # Get SSO Admin Role:
+                paginator = iam.get_paginator('list_roles')
+
+                for page in paginator.paginate():
+                  for role in page['Roles']:
+                    if(f"AWSReservedSSO_{ADMIN_PERMISSION_SET}" in role['Arn']):
+                      arn = role['Arn']
+
+                response_payload['SSOAdminArn'] = arn
+                response_status = cfnresponse.SUCCESS
+
+            except Exception as e:
+              print('ERROR: Caught exception:')
+              print(e)
+              traceback.print_exc()
+
+            finally:
+              print('Sending cfn response')
+              cfnresponse.send(event, context, response_status,
+                              response_payload, physical_resource_id)
+  AdminSSORoleArn:
+    Type: Custom::GetSSOAdminRole
+    DependsOn: GetSSORoleArnLambda
+    Properties:
+      ServiceToken: !Sub arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:Get-Account-AdminSSORoleArn
+
+  ImageEncryptionKMSKey:
+    Type: AWS::KMS::Key
+    Metadata:
+      cfn-lint: { 
+        config: { 
+          ignore_checks: [ EIAMPolicyWildcardResource, EIAMPolicyActionWildcard ],
+          ignore_reasons: [ 
+            EIAMPolicyWildcardResource: "Requires full access to function",
+            EIAMPolicyActionWildcard: "Required in order to properly function"
+          ]
+        }
+      }
+    DependsOn: 
+      - EC2IBRole
+      - AdminSSORoleArn
+    Properties:
+      Description: A Symmetric KMS To Be used for Image Encryption
+      EnableKeyRotation: true
+      KeyPolicy:
+        Version: '2012-10-17'
+        Id: key-default-1
+        Statement:
+          - Sid: Enable IAM User Permissions
+            Effect: Allow
+            Principal:
+              AWS: !Sub "arn:aws:iam::${AWS::AccountId}:root"
+            Action: kms:*
+            Resource: '*'
+          - Sid: Allow administration of the key
+            Effect: Allow
+            Principal:
+              AWS: !GetAtt AdminSSORoleArn.SSOAdminArn
+            Action:
+              - kms:Create*
+              - kms:Describe*
+              - kms:Enable*
+              - kms:List*
+              - kms:Put*
+              - kms:Update*
+              - kms:Revoke*
+              - kms:Disable*
+              - kms:Get*
+              - kms:Delete*
+              - kms:TagResource
+              - kms:UntagResource
+              - kms:ScheduleKeyDeletion
+              - kms:CancelKeyDeletion
+            Resource: '*'
+          - Sid: Allow use of the key
+            Effect: Allow
+            Principal:
+              AWS: 
+                - !Sub "${EC2IBRole.Arn}"
+                - !Sub arn:aws:iam::${AppAccountId}:root
```

Comment:
Instead of hardcoding the `aws` partition into the arn, use the `arn:${AWS::Partition}` pseudo parameter which allows this sample to work in other partitions such as China or AWS GovCloud (US).

For more information on pseudo parameters, see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/pseudo-parameter-reference.html